lib/cogctl/actions/bundles/create.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Cogctl.Actions.Bundles.Create do
+  use Cogctl.Action, "bundles create"
+
+  alias CogApi.HTTP.Client
+
+  @template_extension ".mustache"
+
+  @moduledoc """
+  Creates bundles. The bundles create action creates bundles by taking a bundle
+  config file, running some validations on it and POSTing the resulting json.
+  Special considerations are made for templates. Templates can be inlined in the
+  bundle config or split up into separate files following a specific file hierachy.
+  In the later case this action will read each template and append the contents to
+  the config file.
+
+  The file hierachy looks as follows:
+    templates
+    └── <adapter>
+        └── <template_name>.mustache
+
+  Your template name should correspond to the command that it belongs to. For example,
+  a command called 'date' would have a template called 'date.mustache'
+
+  By default cogctl will look for a directory called 'templates' in the working directory.
+  Optionally you can pass a directory via the '--templates' option.
+  """
+
+  def option_spec do
+    [{:file, :undefined, :undefined, {:string, :undefined}, 'Path to your bundle config file (required)'},
+     {:templates, :undefined, 'templates', {:string, 'templates'}, 'Path to your template directory'}]
+  end
+
+  def run(options, _args, _config, endpoint) do
+    case convert_to_params(options, [file: :required,
+                                     templates: :required]) do
+      {:ok, params} ->
+        with_authentication(endpoint, &do_create(&1, params))
+      :error ->
+        display_arguments_error
+    end
+  end
+
+  defp do_create(endpoint, params) do
+    results = with {:ok, config}         <- Spanner.Config.Parser.read_from_file(params.file),
+                   {:ok, templates}      <- build_template_map(params.templates),
+                   {:ok, amended_config} <- maybe_add_templates(templates, config),
+                   :ok                   <- Spanner.Config.validate(amended_config),
+                 do: Client.bundle_create(endpoint, amended_config)
+
+    case results do
+      {:ok, bundle} ->
+        display_output("Bundle created #{bundle.name}")
+      {:error, message} ->
+        display_error(message)
+    end
+  end
+
+  defp build_template_map(template_dir) do
+    path = Path.relative_to_cwd(template_dir)
+    with {:ok, adapters} <- adapters(path) do
+      templates(path, adapters)
+    end
+  end
+
+  defp adapters(path) do
+    if File.dir?(path) do
+      adapters = File.ls!(path)
+      |> Enum.filter(&File.dir?(Path.join(path, &1)))
+      {:ok, adapters}
+    else
+      {:ok, []}
+    end
+  end
+
+  defp templates(path, adapters) do
+    templates = Enum.reduce(adapters, %{}, fn(adapter, acc) ->
+      Path.join(path, adapter)
+      |> File.ls!
+      |> Enum.filter(&template?/1)
+      |> Enum.reduce(acc, &insert_template(&2, Path.join([path, adapter, &1]), adapter))
+    end)
+    {:ok, templates}
```

Comment:
I'm having some trouble following this nested `Enum.reduce`, but I'm not sure how to rework it. A comment with the final value would probably be helpful though.